### PR TITLE
Wrap host ports metadata in a prefilter

### LINKED ISSUE #86227: Move host ports predicate metadata into pre-filter plugins [Migration Phase 2]
/assign @ahg-g 
/sig scheduling
/priority important-soon

Part of #85822

### PR DESCRIPTION
**Pull request by ahg-g**
**What type of PR is this?**

/kind feature

**What this PR does / why we need it**:
Wraps NodePorts metadata in a prefilter.

**Which issue(s) this PR fixes**:
Fixes #86227

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

/cc @Huang-Wei

### CODE REVIEW

File: pkg/scheduler/framework/plugins/nodeports/node_ports.go [hunk 1]
```diff
@@ -18,31 +18,85 @@ package nodeports
...
+
 // Filter invoked at the filter extension point.
-func (pl *NodePorts) Filter(ctx context.Context, _ *framework.CycleState, pod *v1.Pod, nodeInfo *nodeinfo.NodeInfo) *framework.Status {
-	_, reasons, err := predicates.PodFitsHostPorts(pod, nil, nodeInfo)
+func (pl *NodePorts) Filter(ctx context.Context, cycleState *framework.CycleState, pod *v1.Pod, nodeInfo *nodeinfo.NodeInfo) *framework.Status {
+	state, err := getPreFilterState(cycleState)
+	if err != nil {
+		return framework.NewStatus(framework.Error, err.Error())
+	}
+	_, reasons, err := predicates.PodFitsHostPortsPredicate(pod, []*v1.ContainerPort(state), nodeInfo)
```
Comment (Huang-Wei) [inline pkg/scheduler/framework/plugins/nodeports/node_ports.go]:
```suggestion
	_, reasons, err := predicates.PodFitsHostPortsPredicate(pod, state, nodeInfo)
```

**Review by Huang-Wei — COMMENTED** [review]
LGTM generally. Just a nit.

File: pkg/scheduler/framework/plugins/nodeports/node_ports.go (hunk 1 above)
Comment (ahg-g) [reply pkg/scheduler/framework/plugins/nodeports/node_ports.go]: done.

**Comment by Huang-Wei** [comment]
/lgtm

Thanks @ahg-g .

**Comment by ahg-g** [comment]
@Huang-Wei rebased.

**Pull request merged**